### ISSUE TITLE
Hunspell returns true when checking the spelling of an empty string or a string full of ignored characters

### ISSUE THREAD

**Issue by AhmadAlHallak** [hunspell/hunspell#765]
Currently Hunspell finds empty strings, or strings containing only ignored characters (using the `IGNORE` keyword in the `.aff` file for the language) to be spelled correctly.

This may be the intended behavior, but I can't really understand why that would be desirable, and if it's working as intended for empty strings (for compatibility reasons maybe?), then I don't think that Hunspell should show that a string that is composed solely of ignored characters is spelled correctly (for example, Hunspell finds "ٌٍِ" which is a string composed solely of Arabic diacritical marks which are ignored in the `.aff` for Arabic as correctly spelled, when it shouldn't).

Depending on the intended behavior, this could be rectified by changing the logic around here:

https://github.com/hunspell/hunspell/blob/31e6d6323026a3bef12c5912ce032d88bfef2091/src/hunspell/hunspell.cxx#L496-L497

**Comment by linuxscout** [comment]
> "ٌٍِ"
ٍSalam,
I'm  the maintainer or [Arabic hunspell dictionary](https://github.com/linuxscout/ayaspell).
The IGNORE feature is developed especially for Arabic,
can you explain what's is the main problem for this case.
Thanks

**Comment by AhmadAlHallak** [comment]
Hi Taha, 
Great work on the Arabic dictionary. (big fan of your PyArabic, Mishkal and Qutrub projects)
There is little chance for this to cause problems for end users when Hunspell is used directly as a spellchecker (at least for Arabic), but it can cause problems in cases of Hunspell being used as tool in a program (such as in my case, where I am developing an automatic language switcher). 
Also it's just a bit weird for Hunspell to consider a string full of ignored characters as spelled correctly.

**Comment by linuxscout** [comment]
@AhmadAlHallak, I think that can be fixed by a simple funciton in your code.

**Comment by AhmadAlHallak** [comment]
@linuxscout Yeah, this is what I am currently doing, it's just that I don't feel like this is the correct behavior for these cases. 
I don't see how Hunspell showing that " ٍُِ " is spelled correctly for example can be considered as correct behavior, and that is why I believe that this should be fixed in Hunspell itself rather than at the programs that use it.

The fix is also a bit cumbersome (reading the affix file and extracting the `IGNORE`d characters and then doing a post check after Hunspell to account for this case), while fixing it in Hunspell seems fairly straight forward changing only a couple of lines while adding little to no overhead

**Comment by munzirtaha** [comment]
@AhmadAlHallak:
It's not wrong. Those ignored characters could appear in a text separately in a correct context. e.g.
علامة الرفع الضمة (ُ)
So, if any thing could be correct, don't fix it!

**Comment by AhmadAlHallak** [comment]
> @AhmadAlHallak: It's not wrong. Those ignored characters could appear in a text separately in a correct context. e.g. علامة الرفع الضمة (ُ) So, if any thing could be correct, don't fix it!

Single characters are always considered correct as far as I've seen, but this is not what the issue is about. 

The issue is about Hunspell returning true for any string composed strictly from `IGNORE`d characters in the `.aff` file, for example Hunspell returns true when evaluating the following `  ًََُ  ` in Arabic, which is as correct as `qweqweqweqwe` is in English, which I think we can both agree is not correct in any normal context. So while it's not the wrong behavior in the case you mentioned (and about a dozen or two more), it is wrong in an infinite amount of other cases, which is why this issue exists.

Although I do admit now that I can't immediately think of a fix that takes the current Arabic dictionary and its affixes files in consideration without it being very cumbersome especially with the "Shaddah" being among the `IGNORE`d characters

**Comment by caolanm** [comment]
I think we can live with it as it is
